src/Application/Lockers/Commands/AddLocker/AddLockerCommand.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using Domain.Entities.Physical;
+using Domain.Exceptions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Lockers.Commands.AddLocker;
+
+public record AddLockerCommand : IRequest<LockerDto>
+{
+    public string Name { get; init; }
+    public string Description { get; init; }
+    public string RoomId { get; init; }
+    public int Capacity { get; init; }
+}
+
+public class AddLockerCommandHandler : IRequestHandler<AddLockerCommand, LockerDto>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public AddLockerCommandHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<LockerDto> Handle(AddLockerCommand request, CancellationToken cancellationToken)
+    {
+        var roomGuid = Guid.Parse(request.RoomId);
```

Comment:
If you enforce id should be a guid from the start, this wouldn't have been needed, what if the parsing fail?